File: js/code.js
function hello() {
    console.log("hello world");
}

function demo(parameter) {
    if (parameter) {
        console.log("One");
    } else {
        console.log("Two");
    }
}

function factorial(n) {
    var result = 1;
    for (; n > 0; n--) {
        result = result * n;
    }

    console.log(result);
}

hello()
demo(true)
demo(false)
factorial(10)

File: lua/code.lua
function hello()
    print('hello world')
end

function demo(parameter)
    if parameter then
        print('one')
    else
        print('two')
    end
end

function factorial(n)
    local result = 1
    for i = 1, n do
        result = result * i
    end

    print(result)
end

hello()
demo(true)
demo(0)
demo(false)
factorial(10)